GOLDTracker/src/components/SearchComponent.js

```diff
@@ -1,24 +1,93 @@
+  TouchableOpacity,
+  Animated,
+  Easing,
+  TouchableWithoutFeedback,
+} from "react-native";
 import { SearchBar } from "react-native-elements";
 import Class from "../components/Class";
-import {useNavigation} from "@react-navigation/native";
+import { useNavigation } from "@react-navigation/native";
 import { auth } from "../../firebaseConfig";
-import CategorySearch from "../components/CategorySearch";
 import * as ClassRegister from "./ClassRegister";
+import { COLORS } from "../theme/theme";
+import departmentMapping from "../assets/departmentMapping.json";
+import AntDesign from "react-native-vector-icons/AntDesign";
+import EmptyState from "../components/EmptyState"; // Import EmptyState
 
 const SearchComponent = ({ search, setSearch, setIsSearching, major }) => {
   const [results, setResults] = useState([]);
   const [errorMessage, setErrorMessage] = useState("");
   const [selectedDept, setSelectedDept] = useState(null);
   const [selectedQuarter, setSelectedQuarter] = useState("20251");
   const [isLoading, setIsLoading] = useState(false);
+  const [isDeptDropdownVisible, setIsDeptDropdownVisible] = useState(false);
+  const [isQuarterDropdownVisible, setIsQuarterDropdownVisible] = useState(false);
+  const animatedDeptHeight = useRef(new Animated.Value(0)).current;
+  const animatedQuarterHeight = useRef(new Animated.Value(0)).current;
 
-  const categorySearchRef = useRef(null);
+  const scrollY = useRef(new Animated.Value(0)).current;
   const navigation = useNavigation();
   const API_URL =
     "https://us-central1-goldtracker-beb96.cloudfunctions.net/search";
 
+  // Heights
+  const searchBarHeight = 70; // Adjusted to match actual height
+  const filterBarHeight = 60; // Adjusted to match actual height
+  const totalHeaderHeight = searchBarHeight + filterBarHeight; // 130
+
+  // Interpolation for search bar's animation when scrolling
+  const animatedSearchBarTranslate = scrollY.interpolate({
+    inputRange: [0, searchBarHeight],
+    outputRange: [0, -searchBarHeight],
+    extrapolate: "clamp",
+  });
+
+  // Interpolation for filter bar's animation when scrolling
+  const animatedFilterBarTranslate = scrollY.interpolate({
+    inputRange: [0, searchBarHeight],
+    outputRange: [0, -searchBarHeight],
+    extrapolate: "clamp",
+  });
+
+  // Interpolation for FlatList's paddingTop adjustment
+  const animatedPaddingTop = scrollY.interpolate({
+    inputRange: [0, searchBarHeight],
+    outputRange: [totalHeaderHeight, filterBarHeight], // From 130 to 60
+    extrapolate: "clamp",
+  });
+
+  // Define animatedDropdownTop
+  const animatedDropdownTop = scrollY.interpolate({
+    inputRange: [0, searchBarHeight],
+    outputRange: [searchBarHeight + filterBarHeight, filterBarHeight],
+    extrapolate: "clamp",
+  });
+
+  const departmentOptions = [
+    { code: "Your Major", label: "Cancel Selection" },
+    ...Object.keys(departmentMapping).map((code) => ({
+      code,
+      label: departmentMapping[code][0],
```

Comment:
remove the [0] here to fit the new mapping json. Should be: label: departmentMapping[code]